src/raster_types.jl

```diff
@@ -12,7 +13,24 @@ function Raster(dataset::GDALDatasetH)
                     _getrastercount(dataset)) # number of bands
     finalizer(raster, closeraster)
     raster
-end 
+end
+
+function Base.show(io::IO,raster::Raster)
+     if (raster.dataset == C_NULL)
+         println(io, raster.dataset)
+         return null;
```

Comment:
```
else
         ## TODO: can raster.nband be < 1 when raster.dataset != C_NULL?
end
```
wraster = RasterIO.createraster(path2,10,10,0,Int16,"GTiff") generates an error, so you cannot create a raster with 0 bands. 

But when you manually set the dataset to C_NULL the raster object still exists and you can still query raster.nband. Is this intended for some reason? Perhaps the entire raster object should be destroyed then, or we should find a way to avoid manual change of raster.dataset?